app/ante/reject_msgs.go

```diff
@@ -70,51 +86,40 @@ func (rmd RejectMessagesDecorator) checkMsg(ctx sdk.Context, msg sdk.Msg, nested
 		)
 	}
 
-	if nestedMsgs >= maxNestedMsgs {
-		return fmt.Errorf("found more nested msgs than permitted. Limit is : %d", maxNestedMsgs)
+	typeURL := sdk.MsgTypeURL(msg)
+	for _, pred := range rmd.predicates {
+		if pred(typeURL, depth) {
+			return gerrc.ErrInvalidArgument.Wrapf("disabled: %s", typeURL)
+		}
```

Comment:
could we avoid duplication by moving this after the switch and updating `typeURL` if `MsgGrant`?

```
	typeURL := sdk.MsgTypeURL(msg)

	var (
		err           error
		inner         []sdk.Msg
		authorization authz.Authorization
	)

	switch m := msg.(type) {
	case *authz.MsgExec:
		inner, err = m.GetMessages()
	case *govtypesv1.MsgSubmitProposal:
		inner, err = m.GetMsgs()
	case *group.MsgSubmitProposal:
		inner, err = m.GetMsgs()
	case *authz.MsgGrant:
		if authorization, err = m.GetAuthorization(); err == nil {
			typeURL = authorization.MsgTypeURL()
		}
	default:
	}
	if err != nil {
		return err
	}

	for _, pred := range rmd.predicates {
		if pred(typeURL, depth) {
			return gerrc.ErrInvalidArgument.Wrapf("disabled: %s", typeURL)
		}
	}
```

Reply:
no it doesn't work because you will not check the first time